src/Compilers/CSharp/Portable/FlowAnalysis/NullableWalker.cs

```diff
@@ -1696,25 +1721,69 @@ private static BoundExpression RemoveImplicitConversions(BoundExpression expr)
 
-        // See Binder.GenerateConversionForAssignment for corresponding approach in initial binding.
-        private Result GenerateConversionForAssignment(BoundExpression value, TypeSymbol targetType, Result valueResult)
+        // See Binder.GenerateConversionForAssignment for initial binding.
+        private Result GenerateConversionForAssignment(BoundExpression sourceExpression, TypeSymbolWithAnnotations sourceType, TypeSymbol destinationType)
         {
-            if (targetType is null)
+            if (destinationType is null)
             {
                 return Result.Unset;
             }
-
-            HashSet<DiagnosticInfo> useSiteDiagnostics = null;
-            TypeSymbol sourceType = valueResult.Type?.TypeSymbol;
-            var conversion = (sourceType is null) ?
-                _conversions.ClassifyImplicitConversionFromExpression(value, targetType, ref useSiteDiagnostics) :
-                _conversions.ClassifyImplicitConversionFromType(sourceType, targetType, ref useSiteDiagnostics);
+            var type = sourceType?.TypeSymbol;
+            var conversion = GenerateConversion(_conversions, sourceExpression, type, destinationType);
             if (!conversion.Exists)
             {
-                ReportStaticNullCheckingDiagnostics(ErrorCode.WRN_NullabilityMismatchInAssignment, value.Syntax, GetTypeAsDiagnosticArgument(valueResult.Type?.TypeSymbol), targetType);
+                ReportStaticNullCheckingDiagnostics(ErrorCode.WRN_NullabilityMismatchInAssignment, sourceExpression.Syntax, GetTypeAsDiagnosticArgument(type), destinationType);
             }
 
-            return InferResultNullability(value, conversion, targetType, valueResult.Type);
+            return InferResultNullability(sourceExpression, conversion, destinationType, sourceType);
+        }
+
+        // See Binder.BindNullCoalescingOperator for initial binding.
+        private bool GenerateConversionForConditionalOperator(BoundExpression sourceExpression, TypeSymbol sourceType, TypeSymbol destinationType, bool reportMismatch)
+        {
+            var conversion = GenerateConversion(_conversions, sourceExpression, sourceType, destinationType);
+            bool canConvert = conversion.Exists;
+            if (!canConvert && reportMismatch)
+            {
+                Debug.Assert(GenerateConversion(_conversions.WithNullability(false), sourceExpression, sourceType, destinationType).Exists);
+                ReportStaticNullCheckingDiagnostics(ErrorCode.WRN_NullabilityMismatchInAssignment, sourceExpression.Syntax, GetTypeAsDiagnosticArgument(sourceType), destinationType);
+            }
+            return canConvert;
+        }
+
+        private static Conversion GenerateConversion(Conversions conversions, BoundExpression sourceExpression, TypeSymbol sourceType, TypeSymbol destinationType)
+        {
+            HashSet<DiagnosticInfo> useSiteDiagnostics = null;
+            return UseExpressionForConversion(sourceExpression) ?
```

Comment:
>UseExpressionForConversion(sourceExpression) [](start = 19, length = 44)

This looks too fragile. Can we simply always use ClassifyImplicitConversionFromExpression?

Reply:
The expression may not include the inferred nested nullability.

---
In reply to: [175207063](https://github.com/dotnet/roslyn/pull/25358#discussion_r175207063) [](ancestors = 175207063)

Reply:
> The expression may not include the inferred nested nullability.

I don't see how this condition ensures that this is not going to happen. If that is what we are going after, I would expect the logic targeting that aspect rather than something trying to duplicate logic from ConversionsBase.ClassifyImplicitBuiltInConversionFromExpression 

---
In reply to: [175554389](https://github.com/dotnet/roslyn/pull/25358#discussion_r175554389) [](ancestors = 175554389,175207063)

Reply:
Changed to check `Type` and `ConstantValue` only.

---
In reply to: [175560626](https://github.com/dotnet/roslyn/pull/25358#discussion_r175560626) [](ancestors = 175560626,175554389,175207063)